### ISSUE TITLE
[java] LawOfDemeter - false negative with lambda expression

### ISSUE THREAD

**Issue by tjuwhy** [pmd/pmd#3802]
**Affects PMD Version: 6.42.0**

**Rule:**
 
LawOfDemeter <https://pmd.github.io/latest/pmd_rules_java_design.html#lawofdemeter>

**Description:**  PMD fails to alert LawOfDemeter inside lambda expression.

**Code Sample demonstrating the issue:**

```
import java.util.function.Function;

public class Example {

    public void func() {
        MyObject myObject = MyObjectProvider.get();
        Function<Integer, Integer> f = (i) -> {
            myObject.func(); // false-negative here
            return 1;
        };
    }
}

class MyObject {
    public void func(){
        // do nothing
    }
}

class MyObjectProvider {
    public static MyObject get(){
        return new MyObject();
    }
}

```

**Expected outcome:**

PMD should report a violation at line 8, but doesn't. This is a false-negative.

**Running PMD through:** *[CLI]*

@oowekyala

**Comment by jsotuyod** [comment]
Just as #3801  and for the exact same reasons, this is no longer reproducible on PMD 7
